pkg/nsx/services/common/builder.go

```diff
@@ -0,0 +1,37 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	mpmodel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func QueryTagCondition(resourceType, cluster string) string {
+	return fmt.Sprintf("%s:%s AND tags.scope:%s AND tags.tag:%s",
+		ResourceType, resourceType,
+		strings.Replace(TagScopeCluster, "/", "\\/", -1),
+		strings.Replace(cluster, ":", "\\:", -1))
+}
+
+func ConvertTagsToMPTags(tags []model.Tag) []mpmodel.Tag {
```

Comment:
Since only serviceaccount use it, perhaps it's better to put them in nsxserviceaccount/builder.go ?

Reply:
ConvertTagsToMPTags and ConvertMPTagsToTags is common functions, which might be used by all MP API calls in future. Thus I prefer to put them into common module.